drivers/tee/optee/core.c

```diff
@@ -357,6 +357,26 @@ static bool optee_msg_api_uid_is_optee_api(optee_invoke_fn *invoke_fn)
 	return false;
 }
 
+static void optee_msg_get_os_revision(optee_invoke_fn *invoke_fn)
+{
+	union {
+		struct arm_smccc_res smccc;
+		struct optee_smc_calls_revision_result result;
+	} res = {
+		.result = {
+			.reserved0 = 0
+		}
+	};
+	char id[12] = "";
+
+	invoke_fn(OPTEE_SMC_CALL_GET_OS_REVISION, 0, 0, 0, 0, 0, 0, 0,
+		  &res.smccc);
+
+	if ((u32)res.result.reserved0)
+		snprintf(id, sizeof(id), " (%08x)", (u32)res.result.reserved0);
+	pr_info("revision %lu.%lu%s", res.result.major, res.result.minor, id);
```

Comment:
The stuff with `id` looks a bit more complicated than just:
```
if (res.result.reserved0)
     pr_info("revision %lu.%lu (%08x)", res.result.major, res.result.minor,
             (u32)res.result.reserved0);
else
     pr_info("revision %lu.%lu", res.result.major, res.result.minor);
```

Reply:
Yup will fix